app/models/manageiq/providers/amazon/inventory/persister/shared/cloud_collections.rb

```diff
@@ -0,0 +1,84 @@
+module ManageIQ::Providers::Amazon::Inventory::Persister::Shared::CloudCollections
+  extend ActiveSupport::Concern
+
+  included do
+    # Builder class for Cloud
+    def cloud
+      ::ManagerRefresh::InventoryCollection::Builder::CloudManager
+    end
+
+    def add_miq_templates(extra_properties = {})
+      add_collection(cloud, :miq_templates, extra_properties) do |builder|
+        builder.add_properties(:model_class => ::ManageIQ::Providers::Amazon::CloudManager::Template)
```

Comment:
@kbrock @agrare so instead of options that are nested hash, we have separate method for each (.add_properties, .add_builder_params, etc.). I think l like this aproach. It makes it cleaner and it allows us to redefine individual keys in nested hash, in specific providers. What do you think?

Reply:
more exactly ```add_properties()``` is 'top level' general method.
```add_builder_params()``` and ```add_dependency_attributes()``` are methods for nested hashes (with purpose @ladas wrote above). 
And ```add_targeted_arel()``` is method_missing's substitution for add_properties(:targeted_arel => ...)